python/cudf/cudf/core/column_accessor.py

```diff
@@ -50,6 +108,7 @@ def insert(self, name, value, loc=-1):
         new_keys.insert(loc, name)
         new_values.insert(loc, value)
         self._data = self._data.__class__(zip(new_keys, new_values),)
+        self._clear_cache()
 
     def copy(self):
```

Comment:
Copies are usually deep by default in Pandas land whereas this is a shallow copy from the looks of it, maybe add a `deep=False` here and error if someone passes True?